compiler/fir/checkers/src/org/jetbrains/kotlin/fir/analysis/checkers/expression/FirWhenConditionChecker.kt

```diff
@@ -5,21 +5,37 @@
 
 package org.jetbrains.kotlin.fir.analysis.checkers.expression
 
+import org.jetbrains.kotlin.KtNodeTypes
 import org.jetbrains.kotlin.fir.analysis.checkers.checkCondition
 import org.jetbrains.kotlin.fir.analysis.checkers.context.CheckerContext
 import org.jetbrains.kotlin.fir.analysis.diagnostics.DiagnosticReporter
+import org.jetbrains.kotlin.fir.analysis.diagnostics.FirErrors
+import org.jetbrains.kotlin.fir.analysis.diagnostics.reportOn
 import org.jetbrains.kotlin.fir.analysis.diagnostics.withSuppressedDiagnostics
+import org.jetbrains.kotlin.fir.analysis.getChild
+import org.jetbrains.kotlin.fir.expressions.FirWhenBranch
 import org.jetbrains.kotlin.fir.expressions.FirWhenExpression
 import org.jetbrains.kotlin.fir.expressions.impl.FirElseIfTrueCondition
+import org.jetbrains.kotlin.lexer.KtTokens
 
 object FirWhenConditionChecker : FirWhenExpressionChecker() {
     override fun check(expression: FirWhenExpression, context: CheckerContext, reporter: DiagnosticReporter) {
+        val hasSubject = expression.subject != null
         for (branch in expression.branches) {
             val condition = branch.condition
             if (condition is FirElseIfTrueCondition) continue
             withSuppressedDiagnostics(condition, context) {
                 checkCondition(condition, context, reporter)
             }
+            if (!hasSubject) {
+                checkCommaInBranchCondition(branch, context, reporter)
+            }
+        }
+    }
+
+    private fun checkCommaInBranchCondition(branch: FirWhenBranch, context: CheckerContext, reporter: DiagnosticReporter) {
+        if (branch.source?.elementType == KtNodeTypes.WHEN_ENTRY && branch.source?.getChild(KtTokens.COMMA, depth = 1) != null) {
```

Comment:
Just curious, is there a case where the source for `FirWhenBranch` is _not_ a `WHEN_ENTRY`? Assuming it's a real `when` expression with a subject, and not in `if`.

Reply:
AFAIK `if` is the only case.

Reply:
Ok. I was thinking we don't need the 1st condition here, but it should be ok just to be safe.